docs/concepts/package-types.md

```diff
@@ -0,0 +1,225 @@
+
+```json
+{
+  "name": "<your-package-name>",
+  "private": false,
+  "modular": {
+    "type": "package"
+  },
+  "main": "dist-cjs/index.js",
+  "version": "1.0.0",
+  "module": "dist-es/index.js",
+  "typings": "dist-types/index.d.ts",
+  "dependencies": {},
+  "files": ["dist-cjs", "dist-es", "dist-types", "README.md"]
+}
+```
+
+### When to use the "package" type
+
+Modular `package`s are meant to provide re-usable functionality that can be
+published to third-party registries (like
+[the npm registry](https://www.npmjs.com/)) out-of-the-box; to underline this,
+Modular will, for example, refuse to work with packages that have the
+[`private`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#private)
+field set. You should use packages _only when there is a need to consume their
+build output_; if you just need to share some common source code inside your
+monorepo but you don't need to publish it externally, you should use the
+`source` type instead.
+
+## View
+
+Modular `view`s are Modular `package`s that, by convention, export a default
+React component. They are [created](../commands/add.md) with a default
+entrypoint is `src/index.tsx` in their manifest and they are built and tested
+exactly like `package` types. The only difference is that, since `view`s should
+always export a React component, they can be [`start`ed](../commands/start.md)
+to spawn a local developer server and render their default export to the DOM.
+
+Packages are generated by `modular add` using the
+[`modular-template-view`](https://github.com/jpmorganchase/modular/tree/main/packages/modular-template-view)
+[template](../concepts/templates.md)
+
+## Source
+
+Modular `source` types are packages that contain only source code in their `src`
+directory; they are meant to be imported directly from other packages in the
+monorepository and can be [`test`ed](../commands/test.md), but not
+[`built`](../commands/build.md) or [`start`ed](../commands/start.md). `source`s
+are useful to factor out some common code that's used in a monorepository and
+doesn't need to be built separately or published to a registry. `source`s can
+also be used to temporarily work around
+[circular dependency errors](../concepts/circular-dependencies.md), although
+their use in this sense is limited and it's always prefereable to factor out the
+common parts of a circular dependency in an additional dependency to break the
+cycle.
+
+Packages are generated by `modular add` using the
+[`modular-template-source`](https://github.com/jpmorganchase/modular/tree/main/packages/modular-template-source)
+[template](../concepts/templates.md)
+
+## Template
+
+The Modular `template` type is a special type that is never directly created by
+[`modular add`](../commands/add.md), but it's used by it as a blueprint to
+create other packages of various types. An in-depth explanation of templates can
+be found [here](../concepts/templates.md). Templates are created from scratch;
```

Comment:
If we use a `docs/package-types` structure, then we can merge `docs/package-types/template.md` (this section of this current doc) with the existing `docs/concepts/templates.md`.

Reply:
Good idea, done.